internal/rpc/starknet/types.go

```diff
@@ -172,13 +174,17 @@ func NewTxn(tx types.IsTransaction) (Txn, error) {
 	}
 }
 
-type CommonTxnProperties struct {
-	TxnHash   string   `json:"txn_hash"`
+type BroadcastedTxnCommonProperties struct {
+	Type      TxnType  `json:"type"`
 	MaxFee    string   `json:"max_fee"`
 	Version   string   `json:"version"`
 	Signature []string `json:"signature"`
 	Nonce     string   `json:"nonce"`
-	Type      string   `json:"type"`
+}
+
+type CommonTxnProperties struct {
+	TxnHash string `json:"transaction_hash"`
+	BroadcastedTxnCommonProperties
```

Comment:
Embedded fields should come first.

```suggestion
	BroadcastedTxnCommonProperties
	TxnHash string `json:"transaction_hash"`
```